src/scripts/main.js

```
@@ -1,3 +1,48 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('th');
+  const tbody = table.querySelector('tbody');
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      sortTableByColumn(index);
+    });
+  });
+
+  function sortTableByColumn(columnIndex) {
+    const rowsArray = Array.from(tbody.querySelectorAll('tr'));
+
+    rowsArray.sort((a, b) => {
+      const cellA = a.children[columnIndex].textContent.trim();
+      const cellB = b.children[columnIndex].textContent.trim();
+
+      const isNumericColumn = columnIndex === 2 || columnIndex === 3;
+
+      if (isNumericColumn) {
+        const numA = parseValue(cellA, columnIndex);
+        const numB = parseValue(cellB, columnIndex);
+
+        return numA - numB;
+      } else {
+        return cellA.localeCompare(cellB);
+      }
+    });
+
+    tbody.innerHTML = '';
+    rowsArray.forEach((row) => tbody.appendChild(row));
+  }
+
+  function parseValue(value, columnIndex) {
+    if (columnIndex === 2) {
+      return parseInt(value) || 0;
```

Using `parseInt` here will discard decimal fractions and will not preserve negative signs if they were stripped earlier; also returning `0` on parse failure can incorrectly mix invalid/missing values with real zeros. For numeric columns consider using `parseFloat` and a safer sanitizer for currency/number text (e.g. `value.replace(/[^0-9.-]/g, '')`) and then `parseFloat(...)`. Also, instead of defaulting to `0` on parse failure, consider returning `NaN` and handling `NaN` during comparison so that empty/invalid cells sort consistently (for example, push `NaN` to the end).